modules/components/pages/inputs/redpanda_common.adoc

```diff
@@ -0,0 +1,290 @@
+= redpanda_common
+// tag::single-source[]
+:type: input
+:page-beta: true
+:categories: ["Services"]
+
+component_type_dropdown::[]
+
+
+Consumes data from a Redpanda (Kafka) broker, using credentials from a common `redpanda` configuration block.
+
+ifndef::env-cloud[]
+Introduced in version 4.39.0.
+endif::[]
+
+Use this connector in conjunction with the xref:components:outputs/redpanda_common.adoc[`redpanda_common` output], to read data from one or more topics on a Redpanda cluster, and then write the data to another topic on the same cluster.
+
+To avoid duplicating Redpanda cluster credentials in your `redpanda_common` input, output, or any other components in your data pipeline, you can use a single xref:components:redpanda/about.adoc[`redpanda` configuration block]. For more details, see the <<pipeline-example,Pipeline example>>.
+
+NOTE: If you need to move topic data between Redpanda clusters or other Apache Kafka clusters, consider using the xref:components:inputs/redpanda.adoc[`redpanda` input] and xref:components:outputs/redpanda.adoc[output] instead.
+
+[tabs]
+======
+Common::
++
+--
+
+```yml
+# Common configuration fields, showing default values
+input:
+  label: ""
+  redpanda_common:
+    topics: [] # No default (required)
+    regexp_topics: false
+    consumer_group: "" # No default (optional)
+    auto_replay_nacks: true
+```
+
+--
+Advanced::
++
+--
+
+```yml
+# All configuration fields, showing default values
+input:
+  label: ""
+  redpanda_common:
+    topics: [] # No default (required)
+    regexp_topics: false
+    rack_id: ""
+    start_from_oldest: true
+    fetch_max_bytes: 50MiB
+    fetch_min_bytes: 1B
+    fetch_max_partition_bytes: 1MiB
+    consumer_group: "" # No default (optional)
+    commit_period: 5s
+    partition_buffer_bytes: 1MB
+    auto_replay_nacks: true
+```
+
+--
+======
+
+== Pipeline example
+
+This data pipeline reads data from `topic_A` on a Redpanda cluster, transforms the data, and then writes it to `topic_B` on the same cluster. The cluster details are configured within the `redpanda` configuration block, so you only need to configure them once. This is a useful feature when you have multiple inputs and outputs in the same data pipeline that need to connect to the same cluster.
+
+```
```

Comment:
If my understanding of the use case for this connector is correct, it would be good to add an example using a `redpanda_common` input and output + the config block. Could you provide this?

Reply:
```
input:
  redpanda_common:
    topics: [ foo, bar ]

output:
  redpanda_common:
    topic: baz
    key: ${! @kafka_key }

redpanda:
  seed_brokers: [ "127.0.0.1:9092" ]
  tls:
    enabled: true
  sasl:
    - mechanism: SCRAM-SHA-512
      password: bar
      username: foo
```